cpp/src/arrow/engine/substrait/relation_internal.cc

```diff
@@ -103,13 +104,18 @@ Result<compute::Declaration> FromProto(const substrait::Rel& rel,
               "substrait::ReadRel::LocalFiles::FileOrFiles::format "
               "other than FILE_FORMAT_PARQUET");
         }
-
-        if (!util::string_view{item.uri_file()}.starts_with("file:///")) {
+        if (!util::string_view{item.uri_file()}.starts_with("file:///") &&
+            !util::string_view{item.uri_folder()}.starts_with("file:///") &&
+            !util::string_view{item.uri_path_glob()}.starts_with("file:///")) {
           return Status::NotImplemented(
               "substrait::ReadRel::LocalFiles::FileOrFiles::uri_file "
               "with other than local filesystem (file:///)");
         }
-        auto path = item.uri_file().substr(7);
+
+        std::string path;
+        if(item.has_uri_file()) path = item.uri_file().substr(7);
```

Comment:
Instead of doing two if/else chains on `file/folder/glob` could you combine this one with the one at line 135?